tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,257 @@
+                     client_id="unit-test", clean_session=self.clean_session)
+
+        # Read uuid.ini
+        self.config.read(self.uuid_file)
+
+        edge_system_name = self.config.get('GATEWAY', 'name')
+        local_uuid = self.config.get('GATEWAY', 'local-uuid')
+
+        self.assertEquals(edge_system_name, self.edge_system.name)
+        self.assertEquals(local_uuid, "unit-test")
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_with_valid_msgattr_with_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with valid messaging attributes and non-empty client id
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+
+        # Test with valid messaging attributes
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port, client_id="unit-test", clean_session=self.clean_session,
+                     mqtt_msg_attr=self.mqtt_msg_attr)
+
+        mock_connect.assert_called()
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_with_valid_msgattr_empty_client_id(self, mock_connect):
+        """
+        Test MqttDccComms initialisation with valid messaging attributes with empty client id
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+
+        # Test with valid messaging attributes
+        MqttDccComms(edge_system_name=self.edge_system.name,
+                     url=self.url, port=self.port, client_id="", clean_session=self.clean_session,
+                     mqtt_msg_attr=self.mqtt_msg_attr)
+
+        # Read uuid.ini
+        self.config.read(self.uuid_file)
+
+        edge_system_name = self.config.get('GATEWAY', 'name')
+        local_uuid = self.config.get('GATEWAY', 'local-uuid')
+
+        self.assertEquals(edge_system_name, self.edge_system.name)
+        self.assertIsNotNone(local_uuid)
+        mock_connect.assert_called()
+
+    def test_init_with_invalid_msgattr(self):
+        """
+        Test initialisation with invalid messaging attributes
+        :return:
+        """
+        self.assertRaises(TypeError, lambda: MqttDccComms(edge_system_name=self.edge_system.name,
+                                                          url=self.url, port=self.port,
+                                                          clean_session=self.clean_session, mqtt_msg_attr=""))
+
+    @mock.patch.object(Mqtt, '__init__')
+    def test_connect(self, mock_init):
+        """
+        Test MqttDccComms _connect method
+        :param mock_init: Mocked MQTT init method
+        :return:
+        """
+        mock_init.return_value = None
+
+        client = MqttDccComms(edge_system_name=self.edge_system.name,
+                              url=self.url, port=self.port, clean_session=self.clean_session)
+
+        mock_init.assert_called_with(self.url, self.port, client.identity, client.tls_conf,
```

Comment:
All checks must be `self.` instead of `client.`.  Goal is to check the generated/assigned values.